### PR TITLE
add test for capsule update playbook 

### PR DESCRIPTION
**Pull request by pondrejk**
### Problem Statement
Adjusting to the recent changes in satellite-maintain, for the test's scenario (upgrade from stream to stream) it no longer makes sense to specify the target version.

### Related Issues
https://github.com/SatelliteQE/robottelo/pull/15862 is also needed to make this work, but that change is meant to be cherrypickes, while this should be master only

### CODE REVIEW

File: tests/foreman/api/test_remoteexecution.py
Comment (lpramuk) [inline tests/foreman/api/test_remoteexecution.py:50]: Remove the line completely

**Review by lpramuk — CHANGES_REQUESTED** [review]
Remove `--target-version` completely

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k capsule

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7990
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_remoteexecution.py -k capsule --external-logging
Test Result : ========== 1 failed, 2 deselected, 16 warnings in 1382.00s (0:23:01) ===========
```

File: tests/foreman/api/test_remoteexecution.py
Comment (lpramuk) [reply tests/foreman/api/test_remoteexecution.py]: But there also need to be product changes - user option to specify `upgrade` or `update`

Comment (lpramuk) [reply tests/foreman/api/test_remoteexecution.py]: and `upgrade` is going to always fail for stream/6.16 since there is no 6.17 maintenance bits

File: tests/foreman/api/test_remoteexecution.py
Comment (lpramuk) [inline tests/foreman/api/test_remoteexecution.py:54]: this change is merged already so please rebase

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k capsule

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7996
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_remoteexecution.py -k capsule --external-logging
Test Result : ========== 1 failed, 2 deselected, 16 warnings in 1458.04s (0:24:18) ===========
```

File: tests/foreman/api/test_remoteexecution.py
Comment (lpramuk) [reply tests/foreman/api/test_remoteexecution.py]: According to https://github.com/theforeman/foreman_ansible/pull/727 you should also change `job_template_id` from `upgrade` to `update`  unless you provide next version maintenance repo

File: tests/foreman/api/test_remoteexecution.py
Comment (lpramuk) [inline tests/foreman/api/test_remoteexecution.py:48]: According to https://github.com/theforeman/foreman_ansible/pull/727 you should also change job_template_id from `Capsule Upgrade Playbook` to `Capsule Update Playbook` 
(unless you provide next version maintenance repo in order to upgrade to Y+1 successfully)

**Review by lpramuk — CHANGES_REQUESTED** [review]
Please adjust job template name as well

**Comment by pondrejk** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k capsule
theforeman:
    foreman_ansible: 727
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8004
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_remoteexecution.py -k capsule --external-logging
Test Result : ========== 1 failed, 2 deselected, 15 warnings in 2675.16s (0:44:35) ===========
```

**Comment by pondrejk** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k capsule
theforeman:
    foreman_ansible: 727
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8020
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_remoteexecution.py -k capsule --external-logging
Test Result : ============ 2 deselected, 8 warnings, 1 error in 291.51s (0:04:51) ============
```

**Comment by pondrejk** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k capsule
theforeman:
    foreman_ansible: 727
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8022
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_remoteexecution.py -k capsule --external-logging
Test Result : ========== 1 passed, 2 deselected, 15 warnings in 2774.65s (0:46:14) ===========
```

File: tests/foreman/api/test_remoteexecution.py
Comment (pondrejk) [reply tests/foreman/api/test_remoteexecution.py:71]: Ansible is disabled by default on external capsules

**Review by lpramuk — APPROVED** [review]

**Comment by lpramuk** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k capsule

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8461
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_remoteexecution.py -k capsule --external-logging
Test Result : ========== 1 failed, 2 deselected, 15 warnings in 1457.51s (0:24:17) ===========
```

**Comment by lpramuk** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k capsule
env:
    ROBOTTELO_server__version__release: '6.16.0'
    ROBOTTELO_server__deploy_arguments__deploy_sat_version: '6.16.0'
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8463
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_remoteexecution.py -k capsule --external-logging
Test Result : ================= 2 deselected, 8 warnings, 1 error in 53.70s ==================
```

**Comment by lpramuk** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k capsule
env:
    ROBOTTELO_server__version__release: '6.16.0'
    ROBOTTELO_server__version__snap: '2.0'
    ROBOTTELO_server__deploy_arguments__deploy_sat_version: '6.16.0'
    ROBOTTELO_server__deploy_arguments__deploy_snap_version: '2.0'
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8468
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_remoteexecution.py -k capsule --external-logging
Test Result : ========== 1 failed, 2 deselected, 15 warnings in 1438.00s (0:23:58) ===========
```

**Comment by pondrejk** [comment]
@lpramuk looking at the prt result I suppose that changes from https://github.com/theforeman/foreman_ansible/pull/727 didn't get to 6.16 build... I'm sure they are not in stream either as the last foreman_ansible plugin release was before it got merged. So I guess we can re-run prt after plugin release and stream delivery happens

**Comment by lpramuk** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k capsule

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8500
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_remoteexecution.py -k capsule --external-logging
Test Result : ========== 1 failed, 2 deselected, 15 warnings in 2529.89s (0:42:09) ===========
```

**Comment by lpramuk** [comment]
@pondrejk It is not even in today's stream/6.17 snap

**Comment by lhellebr** [comment]
So what should happen without target version specified? Update to the latest stream snap? Do you really check somewhere that it has really been updated to the LATEST snap?

**Comment by lhellebr** [comment]
Also, should it be in the recent snap?

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k capsule

**Review by pnovotny — APPROVED** [review]
LGTM, PRT pending.

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8868
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/api/test_remoteexecution.py -k test_positive_run_capsule_update_playbook --external-logging
Test Result : ========== 1 failed, 3 deselected, 16 warnings in 1609.68s (0:26:49) ===========
```

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_remoteexecution.py -k test_positive_run_capsule_update_playbook

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 9131
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_remoteexecution.py -k test_positive_run_capsule_update_playbook --external-logging
Test Result : ========== 1 passed, 3 deselected, 16 warnings in 1526.59s (0:25:26) ===========
```

**Pull request merged**
